objects/adffc0/contained/706d9f/contained/4363a8/script.lua

```diff
@@ -0,0 +1,299 @@
+    H7 = {x = 7.68, z = 1.08},
+    H8 = {x = 8.45, z = 6.87},
+}
+
+local thematicPositions = {
+}
+
+function err(color, message)
+    Player[color].broadcast(message, Color.Red)
+end
+
+function castDownCallback()
+    -- If we're not currently casting down, return immediately to avoid overhead.
+    if not Global.getVar("castDown") then
+        return
+    end
+    
+    local board = getObjectFromGUID(Global.getVar("castDown"))
+    local drowningTiles = getBoardTiles(board)
+    for _,guid in pairs(drowningTiles) do
+        local obj = getObjectFromGUID(guid)
+        if obj ~= nil then
+            obj.destruct()
+        end
+    end
+end
+
+-- Starts the Cast Down timer.
+function startCastDownTimer()
+    -- We store the reference to the Cast Down timer in Global, so every Deeps token has access to it.
+    -- Every Deeps token will try to do this, so if it already exists, don't worry about it.
+    if Global.getVar(timerVarName) then
+        return
+    end
+    -- Cast Down lasts 0.5 seconds.
+    -- We need to drop the tiles early, to give things time to fall, so do this every 0.2 seconds.
+    Wait.time(castDownCallback, 0.2, -1)
+    Global.setVar(timerVarName, true)
+end
+
+-- Casts down from the provided location, returning the board found.
+function getBoard(position)
+    local hits = Physics.cast({
+        origin = position, -- + Vector(0,0.2,0) [this was used temporarily while the models were slightly off]
+        direction = Vector(0,-1,0),
+        max_distance = 1,
+    })
+    for _,hit in pairs(hits) do
+        if Global.call("isIslandBoard", {obj=hit.hit_object}) then
+            return hit.hit_object
+        end
+    end
+    return nil
+end
+
+-- Marks a board as having been populated with drowned land tiles.
+-- Does so in a way that persists through saves and reloads.
+function markBoard(board, drowningTiles)
+    local json
+    if board.script_state ~= "" then
+        json = JSON.decode(board.script_state)
+    else
+        json = {}
+    end
+    json.drowningTiles = drowningTiles
+    board.script_state = JSON.encode(json)
+end
+
+-- Returns whether a board has been marked by markBoard.
+function checkBoard(board)
+    if board.script_state == "" then
+        return false
+    end
+    local json = JSON.decode(board.script_state)
+    return json.drowningTiles ~= nil
+end
+
+-- Gets the table of drowning tiles from a board
+function getBoardTiles(board)
+    if board.script_state == "" then
+        return {}
+    end
+    local json = JSON.decode(board.script_state)
+    return json.drowningTiles
+end
+
+-- Places a single drowned land tile, from a bag, into the correct place.
+-- Creates a copy from the bag, 'cause we may need multiple copies for extra bags.
+-- Gives the object a tag to indicate it's submerged in a board.
+function populateTile(bag, guid, position, rotation)
+    -- TODO: Scale this properly according to scaleFactors[SetupChecker.getVar("optionalScaleBoard")].size
```

Comment:
it seems like `calculatePosition` takes into account scaling, so this may already be done?

Reply:
The *position* of the placed tile is adjusted for board scale, but the *scale* of the placed tile isn't. To be honest, I didn't bother dealing with `optionalScaleBoard` properly because that option isn't even exposed to users.